astropy/coordinates/baseframe.py

```diff
@@ -1450,55 +1446,43 @@ def _data_repr(self):
                 ):
                     dif_cls = dif_data.__class__
 
-            else:
-                dif_cls = None
-
             data = self.represent_as(rep_cls, dif_cls, in_frame_units=True)
 
             data_repr = repr(data)
             # Generate the list of component names out of the repr string
             part1, _, remainder = data_repr.partition("(")
-            if remainder != "":
-                comp_str, _, part2 = remainder.partition(")")
-                comp_names = comp_str.split(", ")
+            if remainder:
+                comp_str, part2 = remainder.split(")", 1)
                 # Swap in frame-specific component names
                 invnames = {
                     nmrepr: nmpref
                     for nmpref, nmrepr in self.representation_component_names.items()
                 }
-                for i, name in enumerate(comp_names):
-                    comp_names[i] = invnames.get(name, name)
+                comp_names = (invnames.get(name, name) for name in comp_str.split(", "))
                 # Reassemble the repr string
-                data_repr = part1 + "(" + ", ".join(comp_names) + ")" + part2
+                data_repr = f"{part1}({', '.join(comp_names)}){part2}"
 
         else:
             data = self.data
             data_repr = repr(self.data)
 
-        if data_repr.startswith("<" + data.__class__.__name__):
-            # remove both the leading "<" and the space after the name, as well
-            # as the trailing ">"
```

Comment:
Trusting this comment caused tests to fail, so either the comment was lying or something in our tests is not set up properly.